pkg/controller/controlplane/genericactuator/actuator.go

```diff
@@ -305,13 +309,17 @@ func (a *actuator) reconcileControlPlane(
 	}
 
 	// Apply control plane chart
+	version := extensionscontroller.GetKubernetesVersion(cluster)
+	if cluster.CoreShoot != nil {
```

Comment:
This is already handled by `extensionscontroller.GetKubernetesVersion(cluster)`

Reply:
nice catch